student.rb

```diff
@@ -1,9 +1,12 @@
 require './person'
 
 class Student < Person
+  attr_reader :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
-    super(age, name, parent_permission)
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom.add_student(self)
```

Comment:
- [ ] We need to check if the student we are pushing is not alread in the classroom before adding it to the classroom, in order to avoid unnecessary duplicates, try something like the following: 
```suggestion
    classroom.students.push(self) unless classroom.students.include?(self)
```